src/main/java/Misty/storage/Storage.java

```diff
@@ -0,0 +1,158 @@
+package misty.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import misty.data.TaskList;
+import misty.data.exception.CorruptedFileException;
+import misty.data.task.Deadline;
+import misty.data.task.Event;
+import misty.data.task.Todo;
+
+/**
+ * Saves changes to task list to hard disk as a txt file.
+ */
+public class Storage {
+    private final static  String DEFAULT_DATA_FOLDER_NAME = "data";
+    private final static String DEFAULT_DATA_FILE_NAME = "misty.txt";
+    private String filePath;
+    private File dir = new File(DEFAULT_DATA_FOLDER_NAME);
+    private File dataFile;
+
+    /**
+     * Constructs Storage object.
+     */
+    public Storage() {
+        this.filePath = DEFAULT_DATA_FOLDER_NAME + "/" + DEFAULT_DATA_FILE_NAME;
+        dataFile = new File(filePath);
+    }
+
+    /**
+     * Creates file used to store data.
+     *
+     * @throws IOException If IO error has occurred.
+     * @throws SecurityException If access permission are not given.
+     */
+    public void createFiles() throws IOException, SecurityException {
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        if (!dataFile.exists()) {
+            dataFile.createNewFile();
+        }
+    }
+
+    /**
+     * Writes to save file.
+     *
+     * @param data Data to be written to save file.
+     * @throws IOException If IO error has occurred.
+     */
+    private void writeToFile(String data) throws IOException {
+        FileWriter fileWriter = new FileWriter(dataFile, true);
+        fileWriter.write(data);
+        fileWriter.close();
+    }
+
+    /**
+     * Loads data from save file to task list in chatbot.
+     *
+     * @param taskList TaskList object containing all tasks.
+     * @throws FileNotFoundException If save file is not found.
+     * @throws CorruptedFileException If format of data in save file is incorrect.
+     */
+    public void loadData(TaskList taskList) throws FileNotFoundException, CorruptedFileException {
+        Scanner scanner = new Scanner(dataFile);
+        int taskCount = 0;
+        String[] parameters;
+
+        while (scanner.hasNext()) {
+            String taskData = scanner.nextLine();
+            parameters = taskData.split("\\|");
+            if (parameters[0].contains("T")) {
+                taskList.loadTodo(parameters[2].trim());
+                taskCount++;
+
+                if (parameters[1].contains("1")) {
+                    taskList.loadMark(taskCount);
+                }
```

Comment:
can you extract a method to eliminate one indentation level?

Reply:
Sure prof! Will extract them out from the final submission.